lib/resources/elasticsearch.rb

```diff
@@ -0,0 +1,141 @@
+          .add(:count) { |t, _|
+            t.entries.length
+          }
+
+    filter.connect(self, :nodes)
+
+    attr_reader :nodes
+    def initialize(nodes)
+      @nodes = nodes
+    end
+  end
+
+  class Elasticsearch < Inspec.resource(1)
+    name 'elasticsearch'
+    desc "Use the Elasticsearch InSpec audit resource to test the Elasticsearch
+    configuration settings. The data is extract from the live elastic search
+    cluster using the curl command `curl localhost:9200/_nodes/`. The resource
+    makes uses of FilterTable and its functionality can be used "
+
+    example "
+      describe elasticsearch.nodes do
+        it { should exist }
+      end
+      describe elasticsearch.nodes do
+        its('name') { should include 'my_node' }
+      end
+      describe elasticsearch.nodes do
+          its('name') { should_not include 'node_name' }
+      end
+      elasticsearch.nodes.os.each do |nodeos|
+      	describe nodeos do
+      		its('name'){ should_not cmp 'MacOS' }
+      	end
+      end
+      elasticsearch.nodes.version.each do |node_version|
+      	describe node_version do
+      		it{ should be > '1.2.0' }
+      	end
+      end
+      elasticsearch.nodes.where{os.name == 'Linux' }.settings.each do |node_settings|
+        describe node_settings do
+          its('path.conf') {should cmp '/etc/elasticsearch'}
+        end
+      end
+		  elasticsearch.nodes.where{name == 'package-centos-72' }.settings.each do |node_settings|
+        describe node_settings do
+          its('path.conf') {should cmp '/etc/elasticsearch'}
+        end
+      end
+        "
+
+    def initialize(url = 'http://localhost:9200/_nodes/')
+      return skip_resource 'Package `curl` not avaiable on the host' unless inspec.package('curl').installed?
+      command = inspec.command("curl #{url}")
+      if command.stderr =~ /Failed to connect/
+        return skip_resource 'Connection refused please check ip and port provided'
+      end
+      @content = JSON.parse(command.stdout)
+      return skip_resource 'Invalid node provided' unless !@content['_nodes']['successful'].zero?
+      @nodes = []
+      parse_cluster
+    end
+
+    def nodes
+      ElasticsearchNodeFilter.new(@nodes)
+    end
+
+    def parse_cluster
+      @content['nodes'].each do |node_id, node_data|
+        # resolve Hashie Mash Conflicts
+        node_data.delete('thread_pool')
+        node_data['settings']['http']['type']['default_http_type'] = node_data['settings']['http']['type'].delete('default')
+        node_data['settings']['transport']['type']['default_transport_type'] = node_data['settings']['transport']['type'].delete('default')
+        node_data['settings']['default_path'] = node_data['settings'].delete('default')
+        node = Hashie::Mash.new(node_data)
+        node.node_id = node_id
+        node.plugin_list = plugin_list(node)
+        node.module_list = module_list(node)
+        node.cluster = node.settings.cluster.name
```

Comment:
Suggestion to think about, should this be `cluster_name`?

And for sake of clarity, I would set this from node_data vs. setting it from itself...

`node.cluster_name = node_data['settings']['cluster']['name']`